### PR TITLE
[5.9] Cross-compilation: fix bundles not unpacked on installation

### PR DESCRIPTION
**Pull request by MaxDesiatov**
### Motivation:

`swift experimental-destination install` subcommand works with bundle directories, but fails to unpack bundle archives.

rdar://107367895

### Modifications:

We should pass an instance of an archiver to the installation function to unpack destination bundle archives if needed. Since `withTemporaryDirectory` is `async`, `SwiftDestinationTool` had to be converted to `AsyncParsableCommand` and also gain `@main` attribute for that to work. Additionally, `@main` attribute requires `-parse-as-library` passed in CMake.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Review by neonichu — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by neonichu — APPROVED** [review]

File: Sources/CrossCompilationDestinationsTool/RemoveDestination.swift [hunk 1]
```diff
@@ -17,8 +17,8 @@ import PackageModel
 
 import struct TSCBasic.AbsolutePath
 
-struct RemoveDestination: DestinationCommand {
-    static let configuration = CommandConfiguration(
+public struct RemoveDestination: DestinationCommand {
```
Comment (tomerd) [inline Sources/CrossCompilationDestinationsTool/RemoveDestination.swift:20]: curious why these need to come public?

Comment (MaxDesiatov) [reply Sources/CrossCompilationDestinationsTool/RemoveDestination.swift:20]: Clarified in the PR to the `main` branch, this is due to `SwiftDestinationTool` referencing this type moving to a different target, while `RemoveDestination` could stay where it were.

**Review by tomerd — APPROVED** [review]

**Pull request merged**